src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -350,22 +349,22 @@ private TypeSymbolWithAnnotations(TypeSymbol defaultType, NullableAnnotation nul
             miscellaneousOptions: SymbolDisplayMiscellaneousOptions.UseSpecialTypes | SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier,
             compilerInternalOptions: SymbolDisplayCompilerInternalOptions.IncludeNonNullableTypeModifier);
 
-        internal static TypeSymbolWithAnnotations Create(INonNullTypesContext nonNullTypesContext, TypeSymbol typeSymbol, bool isAnnotated = false, ImmutableArray<CustomModifier> customModifiers = default)
+        internal static TypeSymbolWithAnnotations Create(bool isNullableEnabled, TypeSymbol typeSymbol, bool isAnnotated = false, ImmutableArray<CustomModifier> customModifiers = default)
         {
-            Debug.Assert(nonNullTypesContext != null);
-            Debug.Assert((nonNullTypesContext as Symbol)?.IsDefinition != false);
-#if DEBUG
-            _ = nonNullTypesContext.NonNullTypes; // Should be able to ask this question right away.
-#endif
             if (typeSymbol is null)
             {
                 return default;
             }
 
-            return Create(typeSymbol, nullableAnnotation: isAnnotated ? NullableAnnotation.Annotated : nonNullTypesContext.NonNullTypes == true ? NullableAnnotation.NotAnnotated : NullableAnnotation.Unknown,
+            return Create(typeSymbol, nullableAnnotation: isAnnotated ? NullableAnnotation.Annotated : isNullableEnabled ? NullableAnnotation.NotAnnotated : NullableAnnotation.Unknown,
                           customModifiers.NullToEmpty());
         }
 
+        internal static TypeSymbolWithAnnotations Create(TypeSymbol typeSymbol, ImmutableArray<ModifierInfo<TypeSymbol>> customModifiers)
```

Comment:
>Create [](start = 50, length = 6)

Is this used by `SymbolFactory` only? If so, please move the method back to that class as a helper method to avoid adding the extra overload here, especially since it uses `ModifierInfo<T>`.